### ISSUE TITLE
Ability to set timeframe for /tag/____ pages

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#2439]
We have these great pages to show activity on a topic:

https://publiclab.org/tag/data-logging

![image](https://user-images.githubusercontent.com/24359/37100324-a314ae34-21f0-11e8-8238-f60e164441f6.png)

However, sometimes we want to be able to see this for a given time period, to understand community growth in that timeframe, kind of like on GitHub Pulse:

https://github.com/publiclab/plots2/pulse/monthly vs.

https://github.com/publiclab/plots2/pulse/weekly

That looks like this (so we may need to set an interface, but for now let's just focus on the pages)

![screen shot 2018-03-07 at 10 17 08 am](https://user-images.githubusercontent.com/24359/37100358-bbacf6ea-21f0-11e8-9601-2afcc1875e12.png)

We have these great stats pages where you can set a range: https://publiclab.org/stats/range/10-05-2015/10-05-2016/

The code for that (for example) is:

https://github.com/publiclab/plots2/blob/efc315d3d7bcc08d167c9f6db7816efd805e66f4/app/controllers/stats_controller.rb#L11-L32

## Next steps

This could be solved in a few steps, and the first might be to create a new Tag controller action like:

https://publiclab.org/tag/data-logging?start=16-03-2016&end=14-04-2017

**Comment by rishabhptr** [comment]
@jywarren Can I try this?

**Comment by jywarren** [comment]
Please do! This would be great and @ebarry would like it too.

**Comment by ebarry** [comment]
This will assist with metrics and evaluation, thank you!

**Comment by jywarren** [comment]
Hi @rishabhptr how is this coming? Thanks!

**Comment by rishabhptr** [comment]
Hey, I will make a pr with initial implementation for time ranges in a day or two for you to review. Thanks!

**Comment by rishabhptr** [comment]
Hey, as mentioned in the issue we need to create a new action for this instead of using optional parameters for dates in the original show action for tag. So what would be the response for the example request, the show page which resources filtered within the range or any new page with more descriptive information just like the stats page. Thanks :smile:

**Comment by jywarren** [comment]
I think we can re-use the `show` template but add a line that displays the date range specified. Make sense? Thank you! This is exciting!

**Comment by jywarren** [comment]
And I wonder if we can add an optional filter to the ActiveRecord queries in the same action, rather than making a new action? We could add it after this line, and the matching one from the wildcard section above it:

https://github.com/publiclab/plots2/blob/2f35d8734139a35b524af0c29305122b19bcbffb/app/controllers/tag_controller.rb#L96

We'd want to limit range based on, i guess, node creation date?

**Comment by rishabhptr** [comment]
Yes, that was exactly my thought, instead of new action we can use the same action with optional start and end parameters. Already the show method is quite detailed and good in finding different nodes for a tag and so it would avoid redundancy of the code and layout.
So I think I can go ahead with changes in the controller and then we can add the interface from which user can select and view data for a specific date range. Thanks for the help clarifying the issue! :smile:

**Comment by jywarren** [comment]
super, that sounds great!

**Comment by jywarren** [comment]
We should also plan ahead to make the graph reflect the period of time shown. Looking good!!

**Comment by rishabhptr** [comment]
Yes, it makes sense to update the chart also maybe we can have the graph update while choosing a range from the interface much like we are doing nodes through start and end values. :smile:

**Comment by jywarren** [comment]
I've updated this with a checklist of sub-parts that could be broken out and solved individually. Thanks!!!

**Comment by ViditChitkara** [comment]
This one seems interesting, @rishabhptr are you working on it?

**Comment by jywarren** [comment]
https://github.com/publiclab/plots2/pull/2618 added this nicely -- see:

* https://publiclab.org/tag/spectrometry?start=07-01-2018&end=08-01-2018 versus
* https://publiclab.org/tag/spectrometry

But we need a few more things here --

* [ ] some kind of date selector (see example one already installed at https://publiclab.org/post?legacy=true when checking "This is an event")
* [ ] making the date range apply to the graph
* [ ] making the date range apply to the contributor count
* [ ] making the date range apply to the tag followers count
* [ ] some presets (under a `Period` menu?) like `last week`, `last month` for set time ranges (see some hints for this at https://github.com/publiclab/plots2/blob/master/app/views/stats/range.html.erb)
* [ ] provide link to corresponding period in https://publiclab.org/stats/range/03-01-2018/05-31-2018/

@ViditChitkara what do you think?

**Comment by ViditChitkara** [comment]
@jywarren Great checklist!! Would be happy to implement this step by step!! Thanks!

**Comment by jywarren** [comment]
Adding a new FTO for the range buttons! https://github.com/publiclab/plots2/issues/3298

And the others here could make good FTOs as well 👍

**Comment by jywarren** [comment]
Right now we have this in menu items for "past month" and "past year" -- let's expand on this with more of a full date picker within the same menu!

![image](https://user-images.githubusercontent.com/24359/118716898-5cfdb980-b7f3-11eb-9aee-22f6012b3d5e.png)

**Comment by Manasa2850** [comment]
@jywarren I'm not able to produce this page locally. How do I go about this?

**Comment by jywarren** [comment]
Hi @Manasa2850 do you mean you can't access for example: http://localhost:3000/tag/spectrometry?start=07-01-2018&end=08-01-2018 ?

**Comment by jywarren** [comment]
We may be able to look at #9256 for a quick and standard way to insert dates.
